PascalsTriangle.java

```diff
@@ -0,0 +1,26 @@
+//The approach here is to calculate each number i which is the sum of numbers at index (i-1,j-1) and index (i-1,j) except for the first and last numbers of a row, which are 1 by default
+//Time Complexity: O(n^2)
+//Space Complexity: O(n^2), creating a row for each index
```

Comment:
Please check space complexity. Although you are creating new row everytime , but that row gets added to your result which is your output and output space does not count.